compiler/src/iree/compiler/Codegen/SPIRV/KernelConfig.cpp

```diff
@@ -928,6 +921,30 @@ LogicalResult setCooperativeMatrixConfig(
       subgroupSize = *minSize;
   }
 
+  // Infer if lhs or rhs is transposed to help generate better schedule.
+  auto maps = op.getIndexingMapsArray();
+  OpBuilder b(op);
+  auto lhsMDim = maps[0].getResultPosition(b.getAffineDimExpr(mIndex));
+  auto lhsKDim = maps[0].getResultPosition(b.getAffineDimExpr(kIndex));
+  if (!lhsMDim.has_value() || !lhsKDim.has_value()) {
+    return op.emitError("mDim or kDim not found in LHS indexing map.");
+  }
+  auto rhsKDim = maps[1].getResultPosition(b.getAffineDimExpr(kIndex));
+  auto rhsNDim = maps[1].getResultPosition(b.getAffineDimExpr(nIndex));
+  if (!rhsKDim.has_value() || !rhsNDim.has_value()) {
+    return op.emitError("kDim or nDim not found in RHS indexing map.");
+  }
+  bool transposedLhs = lhsMDim.value() > lhsKDim.value();
+  bool transposedRhs = rhsKDim.value() > rhsNDim.value();
```

Comment:
Slightly simpler could be to just compare `cast<AffineDimExpr>(maps[0].getResults().back()).getPosition()` with `mIndex` and `kIndex`. We can bail out if neither of them are inner most for now (I do not think we've hit that case, and the pipeline more or less assumes that we don't). Then we don't need any calls to `getResultPosition` which scans the whole map.

Reply:
nice idea, thanks :)